packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -220,6 +233,25 @@ function readPackageJson(packageName: string): PackageJson | undefined {
   }
 }
 
+function checkEmptyConfig(
+  userConfig: any,
+  {
+    showSolidityConfigWarnings = false,
+  }: { showSolidityConfigWarnings: boolean } = {
+    showSolidityConfigWarnings: false,
+  }
+) {
+  if (userConfig === undefined || Object.keys(userConfig).length === 0) {
+    let warning = `Hardhat config is returning an empty config object, check the export from the config file if this is unexpected.\n`;
+
+    if (!showSolidityConfigWarnings) {
```

Comment:
Let's add a comment here explaining that this is meant to avoid showing the "learn more..." thing twice.

I don't really like this workaround, but tbh I would've probably done the same thing. Next time we modify something around config and warnings, we should refactor this and pay this technical debt.